provider_group.go

```diff
@@ -21,56 +21,44 @@
 package config
 
 type providerGroup struct {
-	name      string
-	providers []Provider
+	Provider
+	name string
 }
 
 // NewProviderGroup creates a configuration provider from a group of providers.
 // The highest priority provider is the last.
-func NewProviderGroup(name string, providers ...Provider) Provider {
-	return providerGroup{
-		name:      name,
-		providers: providers,
-	}
-}
-
-// Get iterates through the providers and return the value merged from
-// underlying of providers.
-//
 // The merge strategy for two objects
 // from the first provider(A) and the last provider(B) is following:
-// * if A and B are maps, A and B will form a new map with keys from
+// If A and B are maps, A and B will form a new map with keys from
 // A and B and values from B will overwrite values of A. e.g.
 //   A:                B:                 merge(A, B):
 //     keep:A            new:B              keep:A
 //     update:fromA      update:fromB       update:fromB
 //                                          new:B
 //
-// * if A is a map and B is not, this function will panic,
-// e.g. key:value and -slice
+// If A is a map and B is not, this function will return a Value with
+// an error inside.
```

Comment:
It can be an error value from the provider, not a merge error. We construct only correct providers now, no runtime surprises.